textile/chat-features.textile

```
@@ -9,5 +9,148 @@ jump_to:
+** @(CHA-RL5b)@ @[Testable]@ If the operation above fails, then it must be retried after a short wait.
+** @(CHA-RL5c)@ @[Testable]@ If the operation above fails because a channel has entered the @FAILED@ state, then the retry loop must stop and the room must be placed in the @FAILED@ status.
+** @(CHA-RL5d)@ Once all channels have been detached, the room waits until the original channel that caused the retry loop naturally enters the @ATTACHED@ state.
+** @(CHA-RL5e)@ @[Testable]@ If the channel state becomes @FAILED@, then the room status is transitioned to @FAILED@ and the retry loop stops.
+** @(CHA-RL5f)@ @[Testable]@ If the channel state becomes @ATTACHED@, then the room status is transitioned to @ATTACHING@ to begin a new attachment cycle.
+*** @(CHA-RL5f1)@ @[Testable]@ If the attachment cycle succeeds, then the room is transitioned to @ATTACHED@ and any discontinuity errors are broadcast to contributors.
+*** @(CHA-RL5f2)@ @[Testable]@ If the attachment cycle fails because a channel entered @FAILED@, then the room is transitioned to @FAILED@ and the retry loop terminates.
+*** @(CHA-RL5f3)@ @[Testable]@ If the attachment cycle fails because a channel entered @SUSPENDED@, then the room is transitioned to @SUSPENDED@ and the retry loop restarted using the @SUSPENDED@ channel as the subject.
+
+
+h2(#room-configuration). Room Configuration
+
+* @(CHA-RC1)@ Chat Rooms are singleton objects with respect to the Chat Client on which they are created.
+** @(CHA-RC1a)@ @[Testable]@ Requesting a room from the Chat Client will provide an instance of a room with the provided id.
+** @(CHA-RC1b)@ @[Testable]@ Once a room has been created on the Chat Client, assuming it is not subsequently released, then the client will return the same room instance on subsequent calls.
+** @(CHA-RC1c)@ @[Testable]@ If a room is requested with different options to an existing room of the same id, then an error is throw.
```

A specific error?

---

Maybe just ignore the options if the room already exists?

---

Clarified error code.

We elected to throw an `ErrorInfo` rather than ignore, simply because otherwise someone might expect the room to have new options, where that's currently not possible :)